pkg/kp/kv.go

```diff
@@ -111,6 +111,16 @@ func NewConsulStore(opts Options) Store {
 	}
 }
 
+func NewFromClient(client *api.Client) Store {
```

Comment:
I'm super unsure about that, since I do want to pass a consul client that has a certain httpClient. Will revisit after I'm done with refactor though.